### PR TITLE
Nurietzsche/proofs

### PR DESCRIPTION
**Pull request by nurietzsche2021**
Customer form seems to be working just fine. Pay attetion to commented stuff, I left the old implementation

### CODE REVIEW

File: lib/siwapp/commons.ex [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+  #### METATTRIBUTES ####
+
+  @doc """
+  Creates a meta attribute struct with random temp_id by default
+  """
+  def new_meta_attribute() do
+    %MetaAttribute{temp_id: get_temp_id()}
+  end
+
+  def change_metattribute(%MetaAttribute{} = meta_attribute) do
```
Comment (peillis) [inline lib/siwapp/commons.ex:19]: to be consistent with the other name this should be `change_meta_attribute`

File: lib/siwapp/commons.ex [hunk 2]
```diff
@@ -286,4 +303,10 @@ defmodule Siwapp.Commons do
   def change_tax(%Tax{} = tax, attrs \\ %{}) do
     Tax.changeset(tax, attrs)
   end
+
+  #### CALLBACKS ###
```
Comment (peillis) [inline lib/siwapp/commons.ex:307]: These are not callbacks, just private fns

File: lib/siwapp/commons/meta_attribute.ex [hunk 3]
```diff
@@ -0,0 +1,29 @@
+defmodule Siwapp.Commons.MetaAttribute do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  embedded_schema do
+    field :temp_id, :string, virtual: true
+    #   field :delete, :boolean, virtual: true
```
Comment (peillis) [inline lib/siwapp/commons/meta_attribute.ex:7]: Why not remove this? Do you miss it? :)

File: lib/siwapp/commons/meta_attribute.ex [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    field :key, :string
+    field :value, :string
+  end
+
+  def changeset(meta_attribute, attrs) do
+    meta_attribute
+    |> Map.put(:temp_id, meta_attribute.temp_id || attrs["temp_id"])
+    |> cast(attrs, [:key, :value])
+
+    #    |> maybe_mark_for_deletion()
```
Comment (peillis) [inline lib/siwapp/commons/meta_attribute.ex:17]: do not leave this kind of comments

File: lib/siwapp_web/live/customer_live/edit.html.heex [hunk 5]
```diff
@@ -1,10 +1,14 @@
 <h1> <%= @page_title %> </h1>
 
-<.form let={f} for={@customer} phx-submit="save">
+<.form  let={f} for={@changeset} phx-submit= "save", phx-change="validate">
```
Comment (peillis) [inline lib/siwapp_web/live/customer_live/edit.html.heex:3]: do not add spaces where there is no need of them

**Review by peillis — CHANGES_REQUESTED** [review]
warning: function maybe_mark_for_deletion/1 is unused
  lib/siwapp/commons/meta_attribute.ex:20

File: lib/siwapp_web/live/customer_live/form_component.html.heex [hunk 6]
```diff
@@ -1,23 +1,38 @@
...
+  </div>
+<br>
+  <div clas="form-group">
+    <%= label @f, :invoicing_addres, "INVOICING ADDRESS"%>
+    <%= text_input @f, :invoicing_address %>
+    <%= error_tag @f, :invoicing_address %>
+  </div>
+<br>
+  <div clas="form-group">
+    <%= label @f, :shipping_addres, "SHIPPING ADDRESS" %>
```
Comment (carlosescri) [inline lib/siwapp_web/live/customer_live/form_component.html.heex:33]:
```suggestion
    <%= label @f, :shipping_address, "SHIPPING ADDRESS" %>
```

File: lib/siwapp_web/live/customer_live/form_component.html.heex [hunk 7]
```diff
@@ -1,23 +1,38 @@
...
+  </div>
+<br>
+  <div clas="form-group">
+    <%= label @f, :email, "EMAIL" %>
+    <%= text_input @f, :email %>
+    <%= error_tag @f, :email %>
+  </div>
+<br>
+  <div clas="form-group">
+    <%= label @f, :invoicing_addres, "INVOICING ADDRESS"%>
```
Comment (carlosescri) [inline lib/siwapp_web/live/customer_live/form_component.html.heex:27]:
```suggestion
    <%= label @f, :invoicing_address, "INVOICING ADDRESS"%>
```

File: lib/siwapp/commons/meta_attribute.ex [hunk 8]
```diff
@@ -0,0 +1,29 @@
+defmodule Siwapp.Commons.MetaAttribute do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  embedded_schema do
+    field :temp_id, :string, virtual: true
+    #   field :delete, :boolean, virtual: true
+    field :key, :string
```
Comment (carlosescri) [inline lib/siwapp/commons/meta_attribute.ex:8]: For this pattern with a set of unique keys, you could try setting `:key` as the primary key of the embedded schema (I'm almost sure any binary can be a primary key, not only UUIDs). This way you would get rid of the `:temp_id` field. https://hexdocs.pm/ecto/Ecto.Schema.html#module-primary-keys

Comment (carlosescri) [reply lib/siwapp/commons/meta_attribute.ex:8]: forget this @nurietzsche2021

**Pull request closed without merging**